### PR TITLE
generate -> generate_batch

### PR DESCRIPTION
**Pull request by Vaibhavs10**

### CODE REVIEW

File: generate_tp_continuous_batching.py [hunk 1]
```diff
@@ -75,8 +75,8 @@
 print(f"Processing batch of {len(chat_prompts)} prompts...")
 print("=" * 80)
 
-# Generate responses for all prompts in the batch
-outputs = model.generate(**inputs, generation_config=generation_config)
+# Generate responses for all prompts in the batch using continuous batching
+outputs = model.generate_batch(**inputs, generation_config=generation_config)
```
Comment (ArthurZucker) [inline generate_tp_continuous_batching.py]: nice , outputs are gonna be different I think in terms of format (so make sure the decoding it fine!)!

File: generate_tp_continuous_batching.py [hunk 2]
```diff
@@ -41,14 +45,22 @@
     ],
 ]
 
-# Apply chat template to each set of messages
-chat_prompts = [
-    tokenizer.apply_chat_template(messages, tokenize=False)
+# Apply chat template to each set of messages and tokenize in one step
+def tokenize_function(messages):
+    chat_prompt = tokenizer.apply_chat_template(messages, tokenize=False)
```
Comment (ariG23498) [inline generate_tp_continuous_batching.py:50]: Do we need to add `add_generation_prompt =True` here?

**Comment by pcuenca** [comment]
I'm testing with batches of just one sample. What I've seen so far:

* The attention kernel returns very large numbers for the input ids beyond the sequence. In my batch, `max_seqlen_q` is 75, these are the values I get [here](https://github.com/huggingface/transformers/blob/e68146fbe7052a6dc8456f48edabe705dc1f7381/src/transformers/integrations/flash_paged.py#L71) after the first call:

```
attn_output[0][74]
tensor([[-0.5977, -0.5469,  0.3281,  ..., -0.1816,  1.0781, -0.1680],
        [-0.0020,  0.1045, -0.2539,  ..., -0.6523,  0.8047, -0.5820],
        [ 0.2061, -0.2734, -0.1670,  ..., -0.2520,  0.8125, -0.2559],
        ...,
        [-0.0278, -0.0110, -0.0099,  ...,  0.0129,  0.0055,  0.0129],
        [-0.0276, -0.0041, -0.0015,  ...,  0.0135,  0.0035,  0.0031],
        [-0.8828, -0.1992,  0.1436,  ...,  0.4082,  0.2041,  0.3848]], device='cuda:1', dtype=torch.bfloat16)
```


```
attn_output[0][75]
tensor([[-1.3792e+34, -1.1658e-02, -5.2897e-37,  ...,  5.3955e-02, -3.8216e-22, -6.6797e-01],
        [-1.5882e-22,  1.8262e-01, -2.4319e-22,  ...,  1.4746e-01, -3.1543e+35, -1.3184e-01],
        [-3.1543e+35,  3.2959e-02, -3.8216e-22,  ...,  1.7578e-02,  4.2466e-15,  4.1504e-02],
        ...,
        [-1.2930e-36,  2.8992e-03, -5.7672e+06,  ...,  7.3730e-02, -9.2160e+05, -9.6191e-02],
        [ 3.3203e-02,  1.0498e-01, -9.2160e+05,  ..., -2.8931e-02,  5.6946e-30,  5.0049e-02],
        [ 1.3884e-28, -1.8188e-02, -1.8440e-07,  ..., -3.3447e-02,  4.5804e+28,  1.5747e-02]], device='cuda:1', dtype=torch.bfloat16)
```

* I verified that the slice `[:75, :, :]` is correct.
* However, the large numbers beyond that are transformed into `nan`s in the projection.
* After a few steps, all logits are `nan`s.

If I change the attention implementation to use `paged_attention|kernels-community/triton-flash-attn-sink`, I observe the following:
* A single inference run without CB generates reasonable results.
* Passing through the attention interface does not compute large numbers. Logits are not `nan`.
* Results are still gibberish.

So I _think_ there are two issues:
- Large numbers in the attention kernel that should probably be clipped.
- Some interaction between CB and gpt-oss during generation.
